cirq/contrib/quirk/url_to_circuit.py

```diff
@@ -13,22 +13,86 @@
 # limitations under the License.
 import json
 import urllib.parse
-from typing import Any, List, Dict, Optional
+from typing import Any, List, Dict, Optional, Sequence, cast, TYPE_CHECKING, \
+    Iterable
 
-import cirq
+from cirq import devices, circuits, ops
 from cirq.contrib.quirk.cells import (
     Cell,
     CellMaker,
     CellMakerArgs,
     generate_all_quirk_cell_makers,
+    ExplicitOperationsCell,
 )
 
-
-def quirk_url_to_circuit(quirk_url: str) -> 'cirq.Circuit':
+if TYPE_CHECKING:
+    import cirq
+
+
+def quirk_url_to_circuit(
+        quirk_url: str,
+        *,
+        qubits: Optional[Sequence['cirq.Qid']] = None,
+        extra_recognized: Iterable['cirq.contrib.quirk.cells.CellMaker'] = ()
+) -> 'cirq.Circuit':
+    """Parses a Cirq circuit out of a Quirk URL.
+
+    Args:
+        quirk_url: The URL of a bookmarked Quirk circuit. It is not required
+            that the domain be "algassert.com/quirk". The only important part of
+            the URL is the fragment (the part after the #).
+        qubits: Qubits to use in the circuit. The length of the list must be
+            at least the number of qubits in the Quirk circuit (including unused
+            qubits). The maximum number of qubits in a Quirk circuit is 16.
```

Comment:
Question (no action needed): This seems low. What would it take to increase it?

Reply:
When I increase the limit in Quirk, I observe instabilities like running out of video memory and just plain returning the wrong answers. Even at 16 I already have to do gymnastics to avoid precision issues with the frag coord input to the shaders. Basically it would require a lot of careful thinking about what guarantees WebGL can provide.

Reply:
Interesting, thanks!